File: javascript/regex/python/regex.py
import re

alvo = '11a22b33c'
regexp = r'(\d\d)(\w)'
resultado = re.search(regexp, alvo)
print(resultado.group(0))
print(resultado.group(1))
print(resultado.group(2))

# aplicando em loop
resultados = re.finditer(regexp, alvo)
for resultado in resultados:
	print('{} | {} | [{},{}]'
		.format(resultado.group(), resultado.group(1), 
			resultado.start(), resultado.end()))

# utilizando o compilador
exp = re.compile(regexp)
resultado = re.findall(exp, alvo)
print(resultado)


# substituindo valores

alvo = '01-12-2021'
exp = re.compile('-')

resultado = exp.sub('/', alvo)
print(resultado)

File: javascript/regex/javascript/regex.js
const alvo = '11a22b33c'; 
// expressão com string
// const exp = new RegExp('(\\d\\d)(\\w)', 'g');

const exp = /(\d\d)(\w)/g;

// Executa regex uma vez
// const resultado = exp.exec(alvo);
// console.log(resultado);

let resultado = null;
while (resultado = exp.exec(alvo)) {
  console.log(resultado);
  console.log(exp.lastIndex)
}

console.log(exp.test(alvo));


// substituindo valores
let data =  '01-12-2021';
const expData = /-/g;

console.log(data.replace(expData, '/'));

// separando string
const numeros = '100,200-150,200;20';
const expNumeros = /[,;-]/;

console.log(numeros.split(expNumeros));

// Extraindo codigos
const codigo  = 'A121B12112C12212F12G01';
const expCodigo = /[A-Za-z]\d+/g;

console.log(codigo.match(expCodigo));
